runbotics/runbotics-scheduler/src/queue/process/process-file.service.ts

```diff
@@ -49,17 +49,31 @@ export class ProcessFileService {
         if (filenamePrefix.test(base64File)) {
             const fileName = prefix[0].split(':')[1];
             const contentType = prefix[1].split(':')[1];
+            const extension = mime.extension(contentType);
+            if (!extension) {
+                throw new Error('Unable to determine the extension');
+            }
+
             return {
                 fileName,
-                extension: mime.extension(contentType),
-                contentType
+                extension,
+                contentType,
             };
         }
 
         if (standardPrefix.test(base64File)) {
             const contentType = prefix[0].split(':')[1];
             const fileName = randomUUID();
-            return { contentType, extension: mime.extension(contentType), fileName };
+            const extension = mime.extension(contentType);
+            if (!extension) {
+                throw new Error('Unable to determine the extension');
+            }
+
+            return {
+                contentType,
+                extension,
+                fileName,
+            };
         }
```

Comment:
what do you think about mergin check not to repeat the same in 2 ifs?

if (!filenamePrefix.test(base64File) && !standardPrefix.test(base64File)) throw Error...

const fileName = filenamePrefix.test(base64File) ? prefix[0].split(':')[1] randomUUID()
const contentType = prefix[1].split(':')[1];
const extension = mime.extension(contentType);

if (!extension) {
    throw new Error('Unable to determine the extension');
}

return {
    contentType,
    extension,
    fileName,
}

Reply:
In case of `filenamePrefix` check content type is extracted differently `const contentType = prefix[1].split(':')[1];`, but in case of `standardPrefix` it looks like this `const contentType = prefix[0].split(':')[1];`. Small difference but I think it matters.

Reply:
hm, ok. That's fine
I would consider one more line between fileName/contentType then
const contentType = filenamePrefix.test(base64File) ? prefix[1].split(':')[1] : prefix[0].split(':')[1];